evaluation/tasks/auto_task.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+import os
+
+import torch
+
+from evaluation.utils.io import save_json
+
+
+class AutoTask(ABC):
+    def __init__(self, tokenizer, model):
+        self.tokenizer = tokenizer
+        self.model = model
+        self.torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.metrics = {}
+
+    @classmethod
+    def from_task_name(cls, task_name: str, tokenizer, model):
+        all_tasks = cls.__subclasses__()
+        matched_task = [task for task in all_tasks if task.get_display_name() == task_name]
+
+        if not matched_task:
+            raise ValueError(f'Invalid task: {task_name}')
+
+        return matched_task[0](tokenizer=tokenizer, model=model)
+
+    @staticmethod
+    @abstractmethod
+    def get_display_name() -> str:
+        pass
+
+    @abstractmethod
+    def evaluate(self) -> None:
+        pass
+
```

Comment:
We might want a `reconfigure_tokenizer()` that modifies the `tokenizer` for each task, since, for example, some tasks might want to pad to left, others to the right. This method can directly  use the tokenizer parameters in `self.task_args`.